django_rq/queues.py

```diff
@@ -107,12 +107,30 @@ def get_redis_connection(config, use_strict_redis=False):
     return redis_cls(host=config['HOST'], port=config['PORT'], db=config['DB'], password=config.get('PASSWORD', None))
 
 
+# Used as cache to store a Redis client for each queue
+REDIS_CLIENT_BY_QUEUE = {}
+
+
+def get_cached_client_by_queue(name, use_strict_redis=False):
+    """
+    Fetch a Redis client from REDIS_CLIENT_BY_QUEUE. Call get_redis_connection to instantiate a new one if client
+    does not exist yet.
+    """
+    global REDIS_CLIENT_BY_QUEUE
+    from .settings import QUEUES
+    try:
+        client = REDIS_CLIENT_BY_QUEUE[name]
+    except KeyError:
+        client = get_redis_connection(QUEUES[name], use_strict_redis)
+        REDIS_CLIENT_BY_QUEUE[name] = client
+    return client
+
+
 def get_connection(name='default', use_strict_redis=False):
     """
     Returns a Redis connection to use based on parameters in settings.RQ_QUEUES
     """
-    from .settings import QUEUES
-    return get_redis_connection(QUEUES[name], use_strict_redis)
```

Comment:
instead of adding all this code, and im assuming that the new client connection is the issue here -- i havent tested it --why dont you just add a:

```
from django.utils.functional import lru_cache

....
@lru_cache
def get_redis_connection(.....)
     ....
```

and call it a day?